lib/game_teams_manager.rb

```diff
@@ -122,4 +123,68 @@ def fewest_tackles(season)
     end.first)
   end
 
+  def filter_by_teamid(id)
+    @game_teams.select do |game_team|
+      game_team.team_id == id
+    end
+  end
+
+  def game_teams_by_opponent(teamid)
+    filter_by_teamid(teamid).inject({}) do |result, gameteam|
+      if result[@stat_tracker.get_opponent_id(@stat_tracker.get_game(gameteam.game_id), teamid)] == nil
+        result[@stat_tracker.get_opponent_id(@stat_tracker.get_game(gameteam.game_id), teamid)] = [gameteam]
+      else
+        result[@stat_tracker.get_opponent_id(@stat_tracker.get_game(gameteam.game_id), teamid)] << gameteam
+      end
+      result
+    end
+  end
+
+  def game_ids_per_season(season)
+    @stat_tracker.seasonal_game_data[season].map do |games|
+      games.game_id
+    end
+  end
```

Comment:
This is already defined in stat_tracker (see #game_ids_per_season). This can be refactored to just call on that method.

Reply:
I opted to move it out of stat_tracker and call it in gt manager instead